### ISSUE TITLE
On AB+C Portal, Update/Add/Rename EX Folder Subtypes (NN)

### ISSUE THREAD

**Issue by Nadin-Nader** [cityofaustin/atd-data-tech#6021]
**Currently Management is requesting (once or twice a week) for reports on how many requests are received per permit and/or customer type. In order to make it a more efficient and information report, we'd like to update the EX folder subtypes on Portal so that permit applicants are more likely to select the correct permit subtype.**
Also goes with #4469

**Stakeholders:** Paloma, Kim
**DTS Manager:** Nate

**Acceptance Criteria:**
- Changes Apply to the EX folder on AB+C Portal and are stated in the table below:

SubType   Name | SubCode | Request | Original | New
-- | -- | -- | -- | --
Austin Energy Electric | 50693 | Name Change | Austin Energy Electric | Austin Energy (AE)
Austin Transportation | 50694 | Name Change | Austin Transportation | Austin Transportation (ATD)
City Utility AWU/STORM | 50688 | Name Change | City Utility AWU/STORM | Austin Water Utility (AWU)
Austin Energy Chilled Water (OSER) | 50692 | Retire | N/A | N/A
Street/Bridge | 50695 | Retire | N/A | N/A
Corridor Program Office | NEEDED | Add | N/A | Corridor Program Office
Project Connect | NEEDED | Add | N/A | Project Connect
Private Contractors | 50540 | Name Change | Private Contractor | Private Development
Public Works/Electric | 50690 | Name Change | Public Works/Electric | Public Works Department PWD
Watershed Protection Storm Drain | 50696 | Name Change | Watershed Protection Storm Drain | Watershed Protection Department WPD
  |   |   |   |  
  |   |   | New WorkType |  
Franchised Utility | 50685 | Add 3 Worktypes | Gas Utility |  
  |   |   | Other Electric Utility |  
  |   |   | Telecommunication/Cable Utility |

**Comment by Nadin-Nader** [comment]
## **Status:**

Subha is currently working on the back office changes for the portal side in AMANDA

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0138148 (commit - 2b79a93b8a8c) is raised for DEV deployment.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The help text still will have the old names. I suppose that also needs to be changed. Let me know.

![image](https://user-images.githubusercontent.com/51680715/118718460-e1007300-b7ec-11eb-88ee-7e10274b48a9.png)
![image](https://user-images.githubusercontent.com/51680715/118718570-03928c00-b7ed-11eb-880b-2426d8a47ed2.png)

**Comment by SubhaShroff** [comment]
### Status:
1. Waiting on DEV deployment.
2. Waiting for the response on help text question(with screenshots given above).

**Comment by Nadin-Nader** [comment]
@Nadin-Nader ask Paloma /Kim what the Help TEXT SHOULD BE

**Comment by Nadin-Nader** [comment]
@SubhaShroff I spoke to Paloma and she said that the Help Text is still valid. The only thing she asks:
to change the text saying **"Click here to view our permit packets"** to **"Click here to view our permit website."**  

The link is: https://austintexas.gov/department/right-way-row-permits

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Thank you. Sure that change will be done. And I was actually talking about the titles (highlighted in the second screenshot). For 'Austin Energy (AE)', the help text still refers the old name 'Austin Energy Electric'. Likewise, for other subtypes also, it shows the old names.

**Comment by Nadin-Nader** [comment]
@SubhaShroff here is what Paloma said:

**Excavation - Austin Energy**
Austin Energy (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Austin Energy Electric ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our permit webpages.  (And "here" would link to https://austintexas.gov/department/right-way-row-permits.) 

**Excavation - Austin Transportation**
Austin Transportation (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Austin Transportation ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our permit webpages.

**Excavation - Austin Water**
Austin Water (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Austin Water ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our permit webpages.

**Excavation - Franchised Utility**
Franchised Utility (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Franchised Utility ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our permit webpages.

**Then for the Gas subtype:**
Required for removing or otherwise disturbing soil or pavement in the ROW. For Gas Franchised Utilities ONLY.

**For the Electric subtype:**
Required for removing or otherwise disturbing soil or pavement in the ROW. For Electric Franchised Utilities ONLY.

**For the Telecom/Cable subtype:**
Required for removing or otherwise disturbing soil or pavement in the ROW. For Telecom/Cable Franchised Utilities ONLY.

**Excavation - Private Developers**
Private Developers (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Private Contractors ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our permit webpages.

**Excavation - Public Works**
Public Works (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Public Works ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our webpages.

**Excavation - Watershed Protection **
Watershed Protection (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Watershed Protection ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our webpages.

**Excavation - Project Connect**
Project Connect (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Project Connect ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our webpages.

**Excavation - Corridor Program Office**
Corridor Program Office (EX) - Required for removing or otherwise disturbing soil or pavement in the ROW. For Corridor Program Office ONLY.

Excavations greater in length than 300 feet will require an approval through the General Permit Program (974-2000) prior to the issuance of excavation permits.
If this request is related to excavating a driveway approach, curb/gutter or sidewalk for any reason other than utility construction, you will need to apply for a Driveway/Sidewalk permit.

Click here to view our webpages.

**Comment by SubhaShroff** [comment]
@Nadin-Nader The above changes are done and commited - **4e228fc303cc**. Please audit it.

**Comment by Nadin-Nader** [comment]
@SubhaShroff audited the Commit

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader. **SCTASK0139276** - is submitted for DEV deployment of commit-4e228fc303cc.

**Comment by SubhaShroff** [comment]
### YET TO BE edited.......

![image](https://user-images.githubusercontent.com/51680715/119180733-48f5ca00-ba36-11eb-986b-fdfee0bfcbdb.png)

**Comment by SubhaShroff** [comment]
@Nadin-Nader **SCTASK0139988**(commit:0ee04a60133e) is submitted to fix the missing help text for 'Franchised Utility' node.

**Comment by SubhaShroff** [comment]
@Nadin-Nader the commit is deployed in DEV and the help text is added to all nodes now. Below is the screenshot.  Please let me know the feedback.

![image](https://user-images.githubusercontent.com/51680715/119669576-c7b78200-bdfd-11eb-9140-79764ab0aa56.png)

**Comment by SubhaShroff** [comment]
### Status:
Tested today on DEV for the updated/added/removed and respective help text. The screenshot is given above. Ready for testing on DEV.

**Comment by Nadin-Nader** [comment]
@SubhaShroff i was only able to look at a few of them due to the system kicking me off but it looks good. I cannot test any further as my DEV login is corrupt now. Lets go ahead and move this to TEST and then have the stakeholders UAT it.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Submitted TEST deployment request - **SCTASK0140787**  for (commits-2b79a93b8a8c, 4e228fc303cc, 0ee04a60133e).

**Comment by Nadin-Nader** [comment]
##  **Status:**

Waiting on CM Team to deploy to TEST

**Comment by Nadin-Nader** [comment]
##  **Approved for PROD Deployment 🙌🎉🙌🎉**

**Comment by SubhaShroff** [comment]
@Nadin-Nader I submitted PROD deployment request - **SCTASK0140834**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  There is an issue about pushing this to PROD as per Martin. I will try to get in touch with PIPS team in LIT and see how we can handle this.
![image](https://user-images.githubusercontent.com/51680715/120343727-91c34380-c2be-11eb-9e30-dd0ffd066983.png)

**Comment by Nadin-Nader** [comment]
@SubhaShroff i sent a chat to Martin just now to see what can be done or when they think that info will be available in PROD.

**Status:**
Waiting on on Martin's and LIT PIPS Team's response

**Comment by Nadin-Nader** [comment]
@SubhaShroff @TracyLinder Spoke to Martin about this and he said that this could not be deployed as it was making changes to Lookups that were taken out during the roll back of the PIPS changes. He mentioned that this has a minimum of 2 weeks until it can be considered for deployment due to PIPS security concerns.

Martin also informed me that Launch IT's PIPS team are still doing testing and that they still may have another round of testing the security concerns being brought up. Once all of the security concerns have been addressed, they will continue with the deployment of this Particular deployment

**Comment by Nadin-Nader** [comment]
##  **Status:**

On Hold until PIPS Team either addresses all security concerns for deployments or we get a different resolution from the PIPS team on how we can get this in PROD successfully.

Waiting on PIPS'S response

**Comment by Nadin-Nader** [comment]
## **Status:**
Subha spoke with Martin via email and Subha will be working on the single commit for this;
Email from Subha to Martin and their suggestion:

> From: Subhashini Shroff <Subhashini.S@launchitcorp.com> 
> Sent: Wednesday, June 2, 2021 5:26 PM
> To: Aponte, Martin <Martin.Aponte@austintexas.gov>; Rajagopalan, Gita <Gita.Rajagopalan@austintexas.gov>; Thatikonda, Swapna <Swapna.Thatikonda@austintexas.gov>
> Cc: Nader, Nadin <Nadin.Nader@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: RE: V6 PROD deployment - Lookupcode=99061
> 
> Thank you Martin, for the suggestion. Sure, we will work towards making a fresh commit as you suggested.
> 
> From: Aponte, Martin <Martin.Aponte@austintexas.gov> 
> Sent: Wednesday, June 2, 2021 3:04 PM
> To: Subhashini Shroff <Subhashini.S@launchitcorp.com>; Rajagopalan, Gita <Gita.Rajagopalan@austintexas.gov>; Thatikonda, Swapna <Swapna.Thatikonda@austintexas.gov>
> Cc: Nader, Nadin <Nadin.Nader@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: RE: V6 PROD deployment - Lookupcode=99061
> 
> Hello all,
> 
> Most of this is fine but 
> 
> The issue is VALIDLOOKUP_LOOKUPCODE=99061. 
> That lookup seems to have been created for PIPS. 
> 
> But it doesn’t exist on prod since we had to roll back. 
> But there changing it on commit 4e228fc303cc
> 
> The other two commits are okay I think. 
> 
> So you have to have another commit instead of 4e228fc303cc Splitting of the lookup that currently does not exist since we had PIPS rollback.

**Comment by Nadin-Nader** [comment]
Audited and approved commits 48199ba190be, 7bc84a190b1d, 6200e2acf126. 
Next step is to create DP and request the TASK to deploy to PROD

**Comment by SubhaShroff** [comment]
@Nadin-Nader  DEV Deployment request is submitted - SCTASK0143598 (for commits-48199ba190be  -7bc84a190b1d  -6200e2acf126  -0e10ba55123e).

NOTES: These commits are to replace the commit -**4e228fc303cc** that was causing the conflict in PROD as per Martin. Lookupcode=99061 table is totally rolled off due to PIPS roll off, in PROD. So I had to do the above commits to resolve the commit conflict in PROD.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been promoted to DEV by Martin. Please go ahead and UAT the changes and let me know the outcome so we can move this to TEST.

## **Status:**
Dev changes need to be tested to promote to TEST

**Comment by SubhaShroff** [comment]
@Nadin-Nader I tested in DEV and it all looks good.

**Comment by SubhaShroff** [comment]
### Status:
@Nadin-Nader  TEST deployment is done. I tested it. It looks good. Kindly, let me know if we can move #4469 & #6021 to PROD.

![image](https://user-images.githubusercontent.com/51680715/121702750-c5108a00-ca97-11eb-92e2-58da850c83c1.png)

**Comment by TracyLinder** [comment]
### UAT Results
- Tested w/ Izzy today & no bugs found. 

### Approved for PROD

**Comment by SubhaShroff** [comment]
PROD Deployment Request is submitted for #4469 & #6021 - SCTASK0144586.

**Comment by SubhaShroff** [comment]
@TracyLinder @Nadin-Nader PROD Deployment is done. I checked it on PROD Portal. It all looked good.
Needs verification. 🎉

**Comment by SubhaShroff** [comment]
### Status:
PROD deployment is done.

**Comment by Nadin-Nader** [comment]
Email sent to Stakeholders:

> Hi Everyone,
> 
> Hope you are all doing well. I was wanting to let you know that both changes for 4469 and 6021 Update/Add/Remove EX Folder Subtypes are on PROD And ready for UAT. I will keep open for a week or so and if all is well I will close it out on Monday 6/28/21.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services

————————————————————————————

**Comment by SubhaShroff** [comment]
### Status:
On PROD. Waiting to hear from stakeholders.

**Comment by SubhaShroff** [comment]
### Status:
PROD deployment is done, two weeks ago.

**Comment by Nadin-Nader** [comment]
Closing today

**Comment by Nadin-Nader** [comment]
Closing. Good to go Per Kim Just have to add Subtype "Other City Department" (see #6562)
